components/constellation/constellation.rs

```diff
@@ -1945,7 +1955,12 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
                         match diff {
                             SessionHistoryDiff::BrowsingContextDiff { browsing_context_id, ref new_reloader, .. } => {
                                 browsing_context_changes.insert(browsing_context_id, new_reloader.clone());
-                            }
+                            },
+                            SessionHistoryDiff::PipelineDiff { ref pipeline_reloader, new_history_state_id, .. } => {
+                                if let NeedsToReload::No(pipeline_id) = *pipeline_reloader {
+                                    pipeline_changes.insert(pipeline_id, Some(new_history_state_id));
+                                }
```

Comment:
Perhaps add a comment about what to do in the else case? Until we support history state with URLs, I don't think there's anything sensible we can do :(

Reply:
Right, my naive plan is to have another HashMasp 😞 which is a `<PipelineId, ServoUrl>`. It would store the URL that should be used when reloading that Pipeline if different from the normal `LoadData`.